pkg/routeagent_driver/handlers/ovn/non_gateway_route_controller.go

```diff
@@ -0,0 +1,151 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ovn
+
+import (
+	"github.com/pkg/errors"
+	"github.com/submariner-io/admiral/pkg/watcher"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	nodeutil "github.com/submariner-io/submariner/pkg/node"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	clientset "k8s.io/client-go/kubernetes"
+)
+
+type NonGatewayRouteController struct {
+	nonGatewayRouteWatcher watcher.Interface
+	connectionHandler      *ConnectionHandler
+	remoteSubnets          sets.Set[string]
+	stopCh                 chan struct{}
+	transitSwitchIP        string
+	k8sClientSet           clientset.Interface
```

Comment:
This field is unused.

```suggestion
```

Reply:
it is used in line 65

Reply:
The function param is used on that line but it doesn't need to be stored in the struct.

Reply:
Oh  ok , removed.